models/authentication/agent/agent-connected/index.tsx

```diff
@@ -92,7 +93,28 @@ export class AgentConnected {
   }
 
   async getAgentHabilitation() {
-    const superAgentScopes = await superAgentsList.getScopeForAgent(this.email);
+    // Get all groups this agent belongs to from D-Roles API
+    const drolesGroups = await getGroupsByEmail(this.email);
+
+    // Each group has a space-separated string of scopes
+    // Extract all scopes from all groups and flatten into single array
+    const superAgentScopesWithDuplicates = drolesGroups
+      .map((group) => {
+        return group.scopes.split(' ') as IAgentScope[];
+      })
+      .flat();
```

Comment:
All this would be nice into a dedicated model. Maybe a groups model, that is a wrapper around d-roles client